### DISCUSSION TITLE
Dependency Injection and the ActionEventHandler Lifetime

### DISCUSSION THREAD

**Discussion by AHA-Dana** [Nice3point/RevitToolkit#14] [Q&A] [answered]
I'm wondering if there may be a best practice for the ActionEventHandler and it's lifetime. I am able to add a singleton to the Host and retrieve that service in each ViewModel that needs it. On the surface all seems well, however, I'd love to hear if this is acceptable or not.

Other examples I've found using the RevitToolkit seem to create a new instance of the ActionEventHandler when the private field is declared, or when the ViewModel is constructed. If the Host is even suitable for the ActionEventHandler, this leads me to believe adding the handler as transient might be better.

Any input is greatly appreciated!

**Comment by Nice3point** [comment, marked as answer]
it can be Singleton. The only problem is exceptions. If you don't handle exceptions in the catch block, Revit will unload this EventHendler and it will become unusable

**Reply by AHA-Dana** [reply]
Great to know. Thank you for your time!
